pages/common/semgrep.md

```diff
@@ -0,0 +1,17 @@
+# semgrep
+
+> Perform *static source code analysis* (SAST) on your source code to detect security vulnerabilities.
+> Semgrep is a fast SAST tool aiming to minimize false positives.
+> More information: <https://semgrep.dev/>.
+
+- Run semgrep on a source directory, automatically detecting matching rules:
+
+`semgrep --config=auto {{sourcecode_path}}`
+
+- Run semgrep on a source directory, manually selecting the desired rulesets:
+
+`semgrep --config p/{{ruleset_name}} [--config p/{{another_ruleset_name}} ...] {{sourcecode_path}}`
```

Comment:
```suggestion
`semgrep --config p/{{ruleset_name}} [--config p/{{another_ruleset_name}} ...] {{path/to/directory}}`
```
Also, Should command flags be enclosed in `[`?

Reply:
Such `[` usage is unknown for me. Our style guide doesn't mention whether it's allowed or not. It should be explicitly denied or permitted.

Reply:
What it means ist: This is an option that can be repeated any number of times from 0 to n.
I think that is standard man-page syntax, so it should be understandable this way.

Reply:
> I think that is standard man-page syntax, so it should be understandable this way.

This syntax can be proposed to our client specification and style guide. I like it. But for now it's unavailable...

Reply:
How should I change it here? Remove the elipsis? Or remove elipsis and the angular brackets?

Reply:
`p/` is the prefix for the rulesets. The actual values are going to be something like `p/java` or `p/docker`.

Reply:
what about simply

   --config p/{{ruleset_name1}} --config p/{{rulset_name2}}

and assuming that it is auite obvious, that the second instance is optional?

Reply:
Such assumptions should be permitted explicitly by our style guide. ;) Now it [requires](https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md#special-cases) the syntax I suggested.